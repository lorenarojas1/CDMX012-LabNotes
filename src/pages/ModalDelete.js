/* eslint-disable linebreak-style */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable import/prefer-default-export */
import React from 'react';
import {
  deleteDoc,
  doc,
} from 'firebase/firestore';
import { db } from '../firebase-config';

export function ModalDelete({ openModalDelete, closeModalDelete, currentId }) {
  const onDeleteBlog = async (id) => {
    await deleteDoc(doc(db, 'blogs', id));
  };

  if (!openModalDelete) return null;
  return (
    <section className="modalDelete">
      <div className="popupDelete">
        <button
          className="closePopDelete"
          type="button"
          onClick={closeModalDelete}
        >
          X
        </button>
        <p className="titleDelete">
          {' '}
          üóëÔ∏è
          <br />
          Are you sure you want to delete this note?
          {' '}
        </p>
        <div data-testid="btns" className="btnsDelete">
          <button
            id="btnDelete"
            type="button"
            className="btnDelete"
            onClick={() => {
              onDeleteBlog(currentId);
              closeModalDelete();
            }}
          >
            Delete
          </button>
          <button
            name="btn-cancel"
            id="cancel"
            type="button"
            className="btnNoDelete"
            onClick={closeModalDelete}
          >
            Cancel
          </button>
        </div>
      </div>
    </section>
  );
}
