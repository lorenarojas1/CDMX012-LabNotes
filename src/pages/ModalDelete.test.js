import React from 'react';
// import { BrowserRouter as Router } from 'react-router-dom';
import { create, act } from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ModalDelete } from './ModalDelete';

describe(('Test component ModalDelete'), () => {
  test('render component ModalDelete', () => {
    let modalDelete;
    act(() => {
      modalDelete = create(<ModalDelete openModalDelete />);
    });
    expect(modalDelete.toJSON()).toMatchSnapshot();
  });

  const mockOnClick = jest.fn();

  test('value when click on button cancel', () => {
    const componentModal = render(
      <ModalDelete openModalDelete closeModalDelete={mockOnClick} />,
    );
    const button = componentModal.getByText('Cancel');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  test('value when clicked button delete', () => {
    const componentModal = render(
      <ModalDelete openModalDelete closeModalDelete={mockOnClick} />,
    );
    const button = componentModal.getByText('Delete');
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  screen.debug();
});
