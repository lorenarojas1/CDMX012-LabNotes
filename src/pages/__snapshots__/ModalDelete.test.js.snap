// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component ModalDelete render component ModalDelete 1`] = `
<section
  className="modalDelete"
>
  <div
    className="popupDelete"
  >
    <button
      className="closePopDelete"
      type="button"
    >
      X
    </button>
    <p
      className="titleDelete"
    >
      🗑️
      <br />
      Are you sure you want to delete this note?
    </p>
    <div
      className="btnsDelete"
      data-testid="btns"
    >
      <button
        className="btnDelete"
        id="btnDelete"
        onClick={[Function]}
        type="button"
      >
        Delete
      </button>
      <button
        className="btnNoDelete"
        id="cancel"
        name="btn-cancel"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</section>
`;
